contract/models/account_analytic_invoice_line.py

```diff
@@ -61,6 +71,35 @@ class AccountAnalyticInvoiceLine(models.Model):
         help="Sequence of the contract line when displaying contracts",
     )
 
+    @api.depends(
+        'automatic_price',
+        'specific_price',
+        'product_id',
+        'quantity',
+        'analytic_account_id.pricelist_id',
+        'analytic_account_id.partner_id',
+    )
+    def _compute_price_unit(self):
+        """Get the specific price if no auto-price, and the price obtained
+        from the pricelist otherwise.
+        """
+        for line in self:
+            if line.automatic_price:
+                product = line.product_id.with_context(
+                    quantity=line.quantity,
+                    pricelist=line.analytic_account_id.pricelist_id.id,
+                    partner=line.analytic_account_id.partner_id.id,
+                    date=line.env.context.get('old_date', fields.Date.today()),
```

Comment:
All OK when create invoice, but recurring_next_date will be ignored on view.

Reply:
Well, when adding lines, we will see the price for the current date, which is more than acceptable IMO.